src/platform/iotjs_module_i2c-linux-general.inl.h

```diff
@@ -0,0 +1,243 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#ifndef IOTJS_MODULE_I2C_LINUX_GENERAL_INL_H
+#define IOTJS_MODULE_I2C_LINUX_GENERAL_INL_H
+
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <linux/i2c-dev.h>
+
+
+#include "module/iotjs_module_i2c.h"
+
+
+namespace iotjs {
+
+
+int fd;
+int8_t addr;
+
+
+class I2cLinuxGeneral : public I2c {
+ public:
+  explicit I2cLinuxGeneral(JObject& ji2c);
+
+  static I2cLinuxGeneral* GetInstance();
+
+  virtual int SetAddress(int8_t address);
+  virtual int Scan(I2cReqWrap* i2c_req);
+  virtual int Open(I2cReqWrap* i2c_req);
+  virtual int Close();
+  virtual int Write(I2cReqWrap* i2c_rea);
+  virtual int WriteByte(I2cReqWrap* i2c_rea);
+  virtual int WriteBlock(I2cReqWrap* i2c_rea);
+  virtual int Read(I2cReqWrap* i2c_rea);
+  virtual int ReadByte(I2cReqWrap* i2c_rea);
+  virtual int ReadBlock(I2cReqWrap* i2c_rea);
+};
+
+
+I2cLinuxGeneral::I2cLinuxGeneral(JObject& ji2c)
+    : I2c(ji2c) {
+}
+
+
+I2cLinuxGeneral* I2cLinuxGeneral::GetInstance() {
+  return static_cast<I2cLinuxGeneral*>(I2c::GetInstance());
+}
+
+
+JObject makeI2cArray(char* data, int len) {
+  // FIXME: Need to use JS array. (Not implemented)
+  JObject jarray;
+  JObject jlen(len);
+  jarray.SetProperty("length", jlen);
+
+  char index_str[3];
+
+  for (int i = 0; i < len; i++) {
+    sprintf(index_str, "%d", i);
+    JObject v(data[i]);
+    jarray.SetProperty(index_str, v);
+  }
+
+  return jarray;
+}
+
+
+void AfterI2cWork(uv_work_t* work_req, int status) {
+  I2cLinuxGeneral* i2c = I2cLinuxGeneral::GetInstance();
+
+  I2cReqWrap* i2c_req = reinterpret_cast<I2cReqWrap*>(work_req->data);
+  I2cReqData* req_data = i2c_req->req();
+
+  JArgList jargs(2);
+
+  switch (req_data->op) {
+    case kI2cOpOpen:
+    {
+      if (req_data->error == kI2cErrOpen) {
+        JObject error = JObject::Error("Failed to open I2C device.");
+        jargs.Add(error);
+      } else {
+        jargs.Add(JVal::Null());
+      }
+      break;
+    }
+    case kI2cOpReadBlock:
+    {
+      if (req_data->error == kI2cErrReadBlock) {
+        JObject error = JObject::Error("Error reading length of bytes.");
+        jargs.Add(error);
+      } else {
+        jargs.Add(JVal::Null());
+
+        JObject result = makeI2cArray(req_data->buf_data, req_data->buf_len);
+        jargs.Add(result);
+
+        ReleaseBuffer(req_data->buf_data);
+      }
+      break;
+    }
+  }
+
+  MakeCallback(i2c_req->jcallback(), *I2c::GetJI2c(), jargs);
+
+  delete work_req;
+  delete i2c_req;
+}
+
+
+void setAddress(int8_t addr) {
```

Comment:
It will called from another function, and will have more lines in the future.
I'll change the name of the function.